doc/pagination.md

```diff
@@ -0,0 +1,249 @@
+# Pagination
+
+Thunder supports paginated `FieldFunc`s. When creating a paginated field, you
+can either let Thunder handle pagination, or handle pagination yourself.
+
+Our Pagination implementation follows the [Relay spec](https://facebook.github.io/relay/docs/en/pagination-container.html)
+
+## Thunder-managed pagination
+
+The advantage of thunder-managed pagination is that you, as the user, don't have
+to do anything. The downside is that Thunder requires the full collection of
+resource up-front.
+
+```go
+schema := schemabuilder.NewSchema()
+object := schema.Object("user", User{})
+
+// A paginated FieldFunc can take arguments just like a normal FieldFunc
+object.FieldFunc("addresses", func(ctx context.Context, user *User, args struct {}) ([]*Address, error) {
+  // Provide all records to thunder:
+  return GetAddresses(ctx, user, args)
+}, schemabuilder.Paginated)
+```
+
+```graphql
+{
+  user(id: 5) {
+    addresses(
+      # Arguments conforming to the Relay spec.
+      first: 2,
+      after: "",
+			# Also available:
+      # last: $last,
+      # before: $before,
+    ) {
+      totalCount
+      pageInfo {
+        hasNextPage
+        hasPrevPage
+				# Contains pages with page ends and a blank cursor for the first page
+				# If I want to go to page N, I can do `after: pages[n-1]`
+        pages
+        startCursor
+        endCursor
+      }
+      edges {
+        node {
+					# Contains the full Address
+					zipCode
+					# ...
+				}
+				cursor
+      }
+    }
+  }
+}
+```
+
+```json
+{
+	"user": {
+		"addresses": {
+			"totalCount": 50,
+			"pageInfo": {
+				"hasNextPage": true,
+				"hasPrevPage": false,
+				"pages": [
+					"",
+					"Y3Vyc29yMw==",
+				],
+				"startCursor": "Y3Vyc29yMw==",
+				"endCursor": "Y3xyc39ydw==",
+			},
+			"edges": [
+				{
+					"node": {
+						"zipCode": "94103"
+					},
+					"cursor": "Y3Vyc29yMg=="
+        },
+				{
+					"node": {
+						"zipCode": "94104"
+					},
+					"cursor": "Y3Vyc29yMw=="
+        }
+			]
+		}
+	}
+}
+```
+
+### Sorting
+
+You can define sorts for a field in Thunder. The return value of the sort must
+be one of the following (aliases are supported):
+
+- `int`/`int8`/`int16`/`int32`/`int64`
+- `uint`/`uint8`/`uint16`/`uint32`/`uint64`
+- `string`
+
+As you can see below, a sort field does not need to be on the paginated object.
+
+```go
+schema := schemabuilder.NewSchema()
+object := schema.Object("user", User{})
+object.FieldFunc("addresses", func(ctx context.Context, user *User, args struct {}) ([]*Address, error) {
+  return GetAddresses(ctx, user, args)
+}, schemabuilder.Paginated,
+// Each field in this map will be checked when sorting:
+schemabuilder.SortFields{
+	"houseNumber": func(ctx context.Context, a *Address) int64 {
```

Comment:
why would we use houseNumber here instead of the "number" field that will be returned on the address node?  It seems a bit odd that we need to do the mapping between the two field names.

Reply:
I honestly was just trying to come up with a loosely tied (somewhat contrived) example. Would love to swap it out for a better example.

Reply:
^ lmk if you want me to address this